### PR TITLE
Complete template

### LINKED ISSUE #60: Templates Answer
Once you've created a template, you can answer all the questions and fill it out.

### PR DESCRIPTION
**Pull request by haydenrou**
closes #60

### CODE REVIEW

File: models/template.go [hunk 1]
```diff
@@ -28,6 +28,18 @@ type Question struct {
...
+	ID      primitive.ObjectID `bson:"_id,omitempty"`
+	Label   string             `bson:"label"`
+	Type    QuestionType       `bson:"type"`
+	Options []string           `bson:"options,omitempty"`
+	Min     int                `bson:"min,omitempty"`
+	Max     int                `bson:"max,omitempty"`
+	Order   int                `bson:"order"`
+	Error   string             `bson:"error,omitempty"`
+	Value   string             `bson:"value,omitempty"`
+}
```
Comment (haydenrou) [inline models/template.go:41]:
```suggestion
type QuestionView struct {
  Question
}
```

Might save a bunch of boilerplate

File: controllers/templates.go [hunk 2]
```diff
@@ -60,34 +58,83 @@ func Builder() echo.HandlerFunc {
...
+
+		if err != nil {
+			views.AddFlash(c, "Invalid or expired token", views.FlashError)
+
+			return redirect(c, "/login")
+		}
+
+		db.SaveAnswer(database, models.Answer{}.FromForm(template.ID, parsedToken["sub"].(string), questionViews))
+
+		return renderNoBase(c, components.Save("Response to template"))
```
Comment (haydenrou) [inline controllers/templates.go:138]: How will this work if the page is reloaded? We might want to redirect or change the URL here

Comment (haydenrou) [reply controllers/templates.go:123]: Same with the template build creation

Comment (haydenrou) [reply controllers/templates.go:123]: https://github.com/hay-i/gooal/issues/76

**Review by jonnyjones99 — APPROVED** [review]
Looks good

**Pull request merged**